### PR TITLE
0.12

### PR DESCRIPTION
**Pull request by erwanp**
# 0.12.0

Christmas release 🎅  , adding GPU support as well as many new small features and improvements, and an important bug fix for HITRAN users. 

--- 

Changes since [0.11.1](https://github.com/radis/radis/pull/403) (November 2021) : 

## 🔬 [(Physics) Features](https://github.com/radis/radis/issues?q=label%3Aphysics+milestone%3A0.12)

nothing new

## ⚙️[User-interface](https://github.com/radis/radis/issues?q=label%3Ainterface+milestone%3A0.12)

- #380 GPU widgets are released ! 🎉 

https://user-images.githubusercontent.com/16088743/121789972-f34ab280-cbda-11eb-99b0-d7ba019d80d5.mp4

- #414 : support for cross-sections

```
>>> s = radis.test_spectrum()
>>> s.plot("xsection")

COspectrum2502421670720 new quantities added: ['xsection']
```

image

- #413 automatically compute spectral array if possible ; no need to "update".

```
# will always work if possible :
s.take("abscoeff"/"emisscoeff"/"xsection"/"transmittance_noslit"/ etc.)
s.plot("abscoeff"/"emisscoeff"/"xsection"/"transmittance_noslit"/ etc.)
s.get("abscoeff"/"emisscoeff"/"xsection"/"transmittance_noslit"/ etc.)
```

- #412 
- #409 custom user download , requested by @Springder

## 🏃 [Performance improvement](https://github.com/radis/radis/issues?q=label%3Aperformance+milestone%3A0.12)

- #380 unleashes the power of the GPU ! 

## 🐛 [Bug fixes](https://github.com/radis/radis/issues?q=label%3Abug+milestone%3A0.12)

- #416 important : all large-range HITRAN calculations of 0.11 may have had missing lines 
- #408 
- #410

## 📝 [Documentation](https://github.com/radis/radis/issues?q=label%3Adocumentation+milestone%3A0.12)

- #404 
- #289 
- #406

## 🏁 [CI / Dev / Tests](https://github.com/radis/radis/issues?q=label%3Aci-test+milestone%3A0.12)

nothing new

## 🏗️ [Refactor / change in architecture](https://github.com/radis/radis/issues?q=label%3Arefactor+milestone%3A0.12)

nothing new

### CODE REVIEW

**Comment by dcmvdbekerom** [comment]
There's some spectral lines missing in the image of the CO spectrum, is that new?

**Comment by erwanp** [comment]
Well done for noticing it ; it ended up being a very pesky bug due to a combination of new (non-sorted) HITRAN download and the sparse algorithm of 0.11, which hadn't been tested against non-sorted data. Fixed in #416

**Comment by erwanp** [comment]
@dcmvdbekerom I let you merge if the tests pass. New version will be deployed to Pypi automatically after successful merge to the master branch. We may need some hotfix afterwards if GPU parts are not packaged properly

**Review by dcmvdbekerom — APPROVED** [review]

File: radis/test/files/cdsd_hitemp_09_header.txt [hunk 1]
```diff
@@ -2,4 +2,4 @@
  21 2250.000510 2.346E-34 6.522E-01.0666.0652 6907.50990.71-.002959 0.56  4 2 2 02       2 3 3 02 10 1  20  7 1   5  R 75e 1000
  22 2250.000510 9.572E-46 7.015E+02.0712.065212909.52510.83-.005032 0.67  0 1 1 41       0 1 1 31 13 1  66 10 1  38  R120e 1000
  25 2250.000960 2.744E-33 5.749E+02.0689.0786 5507.18620.71-.003741 0.62  0 0 0 31       0 0 0 21  9 1  29  6 1  14  R 52e 1000
- 21 2250.000990 7.019E-36 1.006E-04.0731.0953 5581.66070.68-.002050 0.69  5 1 1 02       2 4 4 02 11 1  26  8 1   8  P 24e  750
\ No newline at end of file
+ 21 2250.000990 7.019E-36 1.006E-04.0731.0953 5581.66070.68-.002050 0.69  5 1 1 02       2 4 4 02 11 1  26  8 1   8  P 24e  750
```
[the remark this answers is not in the document]
Comment (erwanp) [reply radis/test/files/cdsd_hitemp_09_header.txt:5]: same values; new line was added, we're good!

File: radis/test/files/hitran_CO2_fragment.par [hunk 2]
```diff
@@ -5,4 +5,4 @@
  21 3000.441468 1.731E-30 7.072e-04.08710.116 3011.76390.76-.001918       4 1 1 03       0 1 1 11                    R  4e     3677742029 5 4 5 7    11.0    9.0
  23 3000.456018 3.275E-30 4.324e-05.06700.075 1458.78750.74-.003744       0 2 2 11       0 1 1 01                    R 46e     3677642332 5 4 5 7    95.0   93.0
  23 3000.725649 8.155E-28 5.782e-05.06900.092  278.31510.74-.003587       0 1 1 11       0 0 0 01                    R 27e     3677642332 5 4 5 7    57.0   55.0
- 23 3000.761718 1.925E-30 4.029e-05.06670.073 1566.13420.73-.003793       0 2 2 11       0 1 1 01                    R 49f     3677742332 5 4 5 7   101.0   99.0
\ No newline at end of file
+ 23 3000.761718 1.925E-30 4.029e-05.06670.073 1566.13420.73-.003793       0 2 2 11       0 1 1 01                    R 49f     3677742332 5 4 5 7   101.0   99.0
```
[the remark this answers is not in the document]
Comment (erwanp) [reply radis/test/files/hitran_CO2_fragment.par:8]: same values; new line was added, we're good!

File: radis/test/files/hitran_CO_fragment.par [hunk 3]
```diff
@@ -7,4 +7,4 @@
  51    7.479900 8.621E-38 5.746E-08.07480.082 6354.17910.75-.000310              3              3                    R  1      477665 5 8 2 2 1 7     5.0    3.0
  51    7.549901 7.177E-33 1.867E-07.07480.082 4263.83720.75-.000310              2              2                    R  1      677665 5 8 2 2 1 7     5.0    3.0
  51    7.619908 4.436E-28 3.961E-07.07480.082 2147.08110.75-.000310              1              1                    R  1      687665 5 8 2 2 1 7     5.0    3.0
- 51    7.689920 2.566E-23 6.914E-07.07480.082    3.84500.75-.000310              0              0                    R  1      687665 5 8 2 2 1 7     5.0    3.0
\ No newline at end of file
+ 51    7.689920 2.566E-23 6.914E-07.07480.082    3.84500.75-.000310              0              0                    R  1      687665 5 8 2 2 1 7     5.0    3.0
```
[the remark this answers is not in the document]
Comment (erwanp) [reply radis/test/files/hitran_CO_fragment.par:10]: same values; new line was added, we're good!

**Pull request merged**